include/azure_credentials_provider.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#ifdef LINK_AZURE_LIBS
+
+#include "api_status.h"
+#include "configuration.h"
+#include "future_compat.h"
+
+#include "err_constants.h"
+#include "future_compat.h"
+
+#include <azure/core/datetime.hpp>
+#include <chrono>
+#include <mutex>
+// These are needed because azure does a bad time conversion
+#include <exception>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <memory>
+
+#  ifdef LINK_AZURE_LIBS
+#    include <azure/core/credentials/credentials.hpp>
+#  endif
+
+namespace reinforcement_learning
+{
+
+template<typename T>
+class azure_credentials_provider
+{
+public:
+  template<typename... Args>
+  azure_credentials_provider(Args&&... args) :
+    _creds(std::make_unique<T>(std::forward<Args>(args)...)) {}
+
+  int get_credentials(const std::vector<std::string>& scopes, std::string& token_out,
+      std::chrono::system_clock::time_point& expiry_out)
```

Comment:
needs api_status ?  maybe

Reply:
api_status is reported in the http client wrapper; however, i added the tracer instead.